standards/bulk.md

```diff
@@ -0,0 +1,174 @@
+```json
+// RESPONSE
+HTTP/1.1 400
+Content-Type: application/problem+json
+{
+    "title": "Invalid Data",
+    "status": 400,
+    "detail": "Operations collection may only contain a maximum of '100' actions per request.",
+    "instance": "/articles",
+    "requestId": "b6d9a290-9f20-465b-bcd3-4a5166eeb3d7"
+}
+```
+
+#### Request
+
+- Bulk operations **MUST** include a request body schema 
+```json
+{
+    "transaction": "ATOMIC" | "ISOLATED",                                   // default: "ATOMIC", optional. transactionality of the request
+    "operations": [                                                         
+        {                                                                   
+            "operationId": "string" | null,                                 // optional consumer generated id to associate with the operation for comparison to result.
+            "action": "CREATE" | "UPDATE" | "CREATE_UPDATE" | "DELETE",     // indicate intent of operation (can use subset, but do not extend)
+            "ifMatch": "string" | null,                                     // if-match is an optional ETag that can be passed for optimistic concurrency
+            "entity": {
+                ... MATCHING ENTITY...                                      // must match entity schema resource from the collection
+            }                                                               // schema on entity is static and not dynamic
+        },
+    ]
+}
+```
+
+- Bulk operation requests **MAY** be designated as `ATOMIC` or `ISOLATED` via the `transaction` field.
+    - The default transaction type **MUST** be `ISOLATED`.
+    - The transaction type **MAY** be optionally left out of the request schema in implementation.
+    - `ATOMIC` transactions will either succeed or fail together.
+    - `ISOLATED` transactions will allow for individual operations to succeed or fail independently.
+- `operationId` **MAY** be used to associate the operation in the request with the resulting operation in the response. This is useful for tracking the outcome of each operation in the response where it might be ambiguous to reference via `entityId`.
+- `action` enumeration **MUST NOT** be extended, but **MAY** be a subset of the enumeration values where not all actions are required.
+- `etag` **MUST** be used supported in the request schema if it is used for other RESTful operations on the same resource.
+- Operation collections in a request **MAY** include entities of the same ID, in which case the operations are performed in the order they are received. In this case the use of `ETag` may not work as expected for subsequent concurrency checks. Operation ID should be used by consumer in this case to identify the outcome of each operation, especially if transaction type is `ISOLATED`.
```

Comment:
I can live with this if this is the consensus and there's a use case for it. But it feels a little bad that allowing this is at odds with other request inputs and can make things a bit ambiguous.

What is the expected behavior if I give `ifMatch` values and the same entity multiple times? Does the API just ignore the `ifMatch`? Does the first succeed and the 2nd always fail? It seems like it would be cleaner if we just avoided this whole situation, but maybe it's necessary to support multiple updates instead of the client merging it themselves ahead of time?

Reply:
Based on prior conversation, the consensus did seem to be to allow for this. That being said, its always easier to NOT allow it now, and then allow it later from a compatibility perspective. I'm more inclined as you say to indicate its not allowed to keep things "clean". Based on your comment, I'm going to change this to indicate its not allowed. We can change it later if a use case presents itself that is beyond the theoritical.